[BE][flatbuffer] Remove code duplications and refactor

qihqi opened the pull request:
Summary:
Remove code dup in import.cpp / export_modules.cpp such that
1. Only one copy of switching logic (detect flatbuffer / is_flatbuffer);
2. Move detection of includeness of flatbuffer to runtime (so no more macros)

This also reverts the dependency of import.cpp -> flatbuffer_loader.cpp to flatbuffer_loader.cpp -> import.cpp.

Differential Revision: D36926217

zhxchen17 reviewed (approved):
please run lintrunner

qihqi committed:
[BE][flatbuffer] Remove code duplications and refactor (#79184)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/79184

Remove code dup in import.cpp / export_modules.cpp such that
1. Only one copy of switching logic (detect flatbuffer / is_flatbuffer);
2. Move detection of includeness of flatbuffer to runtime (so no more macros)

This also reverts the dependency of import.cpp -> flatbuffer_loader.cpp to flatbuffer_loader.cpp -> import.cpp.

Test Plan: uni ttests

Reviewed By: zhxchen17

Differential Revision: D36926217

fbshipit-source-id: 1f3504f0742d52b16ad9835c42c83e2f9d0bfd55
